packages/amplication-client/src/Application/code-view/CodeViewPage.tsx

```diff
@@ -0,0 +1,66 @@
+import { useQuery } from "@apollo/client";
+import React, { useState } from "react";
+import { match } from "react-router-dom";
+import { App } from "../../models";
+import { GET_APP_GIT_REPOSITORY } from "../git/SyncWithGithubPage";
+import CodeViewBar from "./CodeViewBar";
+import CodeViewEditor from "./CodeViewEditor";
+import useNavigationTabs from "../../Layout/UseNavigationTabs";
+import PageContent from "../../Layout/PageContent";
+
+import "./CodeViewPage.scss";
+
+const CLASS_NAME = "code-view-page";
+const NAVIGATION_KEY = "CODE_VIEW";
+type Props = {
+  match: match<{ application: string }>;
+};
+
+export type CommitListItem = {
+  id: string;
+  name: string;
+};
+
+export type FileDetails = {
+  buildId: string;
+  filePath: string;
+  isFile: boolean;
+  fileName: string;
+};
+
+function CodeViewPage({ match }: Props) {
+  const applicationId = match.params.application;
+  const [fileDetails, setFileDetails] = useState<FileDetails | null>(null);
+  useNavigationTabs(applicationId, NAVIGATION_KEY, match.url, "Code View");
```

Comment:
I think this mechanism should be decoupled from end point component and should be at the top level of the routing or app. it is not for this PR, but I think it will prevent performance issue in future